src/lucky/renderable.cr

```diff
@@ -103,19 +103,66 @@ module Lucky::Renderable
     response.print
   end
 
+  private def handle_response(_response : Nil) forall T
+    # This is a hack to force Crystal to regenerate this method for
+    # each action. That way @type refers to the action with the problem,
+    # and not the first action to be compiled.
+    {% @type %}
+    {%
+      raise <<-ERROR
+
+
+      #{@type} returned Nil
+
+      But it should return a Lucky::Response.
+
+      Try this...
+
+        ▸ Ensure the action returns a response with html/redirect/json/etc.
+        ▸ Ensure all conditionals (like if/else) return a response with html/redirect/json/etc.
+
+      For example...
+
+        get "/admin/users" do
+          # Make sure there is a response in all conditional branches
+          if current_user.admin?
+            html IndexPage, users: UserQuery.new
+          else
+            redirect Home::Index
+          end
+        end
```

Comment:
Added a specific example for when there is a nil returned

Reply:
Love it

Reply:
Same ❤️